app/src/main/java/com/karumi/data/repository/MemorySuperHeroDataSource.kt

```diff
@@ -0,0 +1,34 @@
+package com.karumi.data.repository
+
+import com.karumi.common.TimeProvider
+import com.karumi.domain.model.DomainError
+import com.karumi.domain.model.NotIndexFoundDomainError
+import com.karumi.domain.model.SuperHero
+import org.funktionale.either.Either
+import java.util.concurrent.TimeUnit
+
+class MemorySuperHeroDataSource(private val timeProvider: TimeProvider) : SuperHeroDataSource {
+    companion object {
+        private val TIME_UPDATE = TimeUnit.SECONDS.toMillis(1)
+    }
+
+    private val cache = LinkedHashMap<String, SuperHero>()
+    private var lastUpdate = 0L
+
+    override fun get(key: String): Either<DomainError, SuperHero> =
+        cache[key]?.let { Either.right(it) }
+            ?: Either.left(NotIndexFoundDomainError(key))
+
+    override fun isUpdated(): Boolean = timeProvider.time - lastUpdate < TIME_UPDATE
```

Comment:
Shouldn't we expect a `key` parameter to see if the stored value is expired or not?

Reply:
I'm expiring the cache all together for simplifying the use case. I can change it if you think if the project will be more clear and useful

Reply:
No need if you already thought about it :)